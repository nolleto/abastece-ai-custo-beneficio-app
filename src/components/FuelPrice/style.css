.text {
  margin: 40px 10px 20px 10px;
  font-size: 20px;
}
