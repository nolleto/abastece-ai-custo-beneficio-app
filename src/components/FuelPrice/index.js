import React from 'react';
import './style.css';

const FuelPrice = ({ valorCombustivel, valorInteiroPago, valorRealPago}) => {
  const hasValue = value => value > 0;
  const hasValues = (...values) => values.every(hasValue)
  const hasValorRealCombustivel = hasValues(valorCombustivel, valorInteiroPago, valorRealPago);
  const getLitrosAbastecidos = () => valorInteiroPago / valorCombustivel;
  const getValorRealCombustivel = () => (valorRealPago / getLitrosAbastecidos()).toFixed(2);

  if (!hasValorRealCombustivel) {
    return (
      <p className='text'>
        Preencha os campos acima para obter o valor do combustível.
      </p>
    );
  }

  return (
    <p className='text'>
      O valor do combustível é de R$ {getValorRealCombustivel()}
    </p>
  );
};

export default FuelPrice;
