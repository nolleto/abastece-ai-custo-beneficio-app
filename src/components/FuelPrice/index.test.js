import React from 'react';
import FuelPrice from './index';
import { shallow } from 'enzyme';

describe('FuelPrice component', () => {
  it('matches snapshot when has no values', () => {
    const snapshot = shallow(<FuelPrice />);

    expect(snapshot).toMatchSnapshot();
  });

  it('matches snapshot when has all values', () => {
    const snapshot = shallow(
      <FuelPrice
        fuelPrice={2}
        requestPrice={50}
        realPricePaid={40}
      />
    );

    expect(snapshot).toMatchSnapshot();
  });
});
