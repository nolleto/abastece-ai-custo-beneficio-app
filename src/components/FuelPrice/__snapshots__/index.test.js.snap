// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FuelPrice component matches snapshot when has all values 1`] = `
<p
  className="text"
>
  O valor do combustível é de R$ 
  1.60
   (
  25.00
   litros abastecidos)
</p>
`;

exports[`FuelPrice component matches snapshot when has no values 1`] = `
<p
  className="text"
>
  Preencha os campos acima para obter o valor do combustível.
</p>
`;
